### PR TITLE
Remove unnecessary Collection methods and add Collection#partition

### PR DESCRIPTION
**Pull request by 1Computer1**
**Please describe the changes this PR makes and why it should be merged:**

This PR reworks Collection a bit.
- Removed all methods `Collection#xxxArray` since the same can be done with `Array.from` or `Collection#array`.
- Removed the (prop, value) pattern you can use on `Collection#find`.
- Which also means `Collection#exists`  and `Collection#findAll` are removed.
- Added `Collection#partition`.
- Reworked some documentation and added more examples.

**Semantic versioning classification:**  
- [X] This PR changes the library's interface (methods or parameters added)
  - [X] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by dragonfire535** [comment]
> Removed all methods Collection#xxxArray since the same can be done with Array.from or Collection#array.

This is an okay change, the xxxArray methods are basically shortcuts anyway.

> That also means Collection#map now returns a Collection rather than an Array. This is a good change since now you can chain Collection or Array methods rather than just Array methods.

This is not okay, and will break _tons_ of existing functionality. `map` is supposed to map out the values in the collection to an array, not give the collection back, that doesn't make sense.

> Removed Collection#findAll and merged its behavior with Collection#filter.

Okay 👍 

> Removed all thisArg parameters. They are never used within the source anyways. For those who wants to keep consistency with the Array methods, note that Collection#find did not have thisArg in the first place, and that it is inconsistent to have some methods that have it and some that do not.

It'd be better to make `Collection#find` not have the double-functionality in the first place.

> Added Collection#partition.

I can see the use-case.

**Comment by ShayBox** [comment]
What about partition#defragment  🤣

**Comment by 1Computer1** [comment]
> It'd be better to make Collection#find not have the double-functionality in the first place.
For consistency, that would mean removing the `findAll` functionality from `filter`.
I'm thinking of just removing the (prop, value) behavior entirely.

**Comment by 1Computer1** [comment]
I've reverted map returning a collection. After some thinking it seems that its usecase isn't as important as breaking all the code. It's also really unpopular with people too.

File: src/util/Collection.js [hunk 1]
```diff
@@ -277,73 +213,80 @@ class Collection extends Map {
...
 
   /**
-   * Identical to
+   * Applies a function to produce a single value. Identical in behavior to
    * [Array.reduce()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce).
    * @param {Function} fn Function used to reduce, taking four arguments; `accumulator`, `currentValue`, `currentKey`,
    * and `collection`
    * @param {*} [initialValue] Starting value for the accumulator
    * @returns {*}
+   * @example guilds.reduce((acc, guild) => acc + guild.memberCount);
```
Comment (bdistin) [inline src/util/Collection.js]: This example needs an initialValue of 0 to work. Else the first time the callback is called, acc is the first Guild object, not its memberCount.

**Comment by iCrawl** [comment]
Rebase @1Computer1

**Pull request merged**